applications/client/modules/procedureapi.py

```diff
@@ -6,18 +6,44 @@ class ProcedureApi():
     def __init__(self, procedure_name):
         self.procedure_name = procedure_name
 
-    def write_value(self, name, data):
-        pass
+    def write_value(self, dictionary):
+        """ Writes key value pairs into the values table
+        param dictionary: Takes dictionary of key/value pairs as input
+        """
+        db = current.db
+        for key in dictionary.keys():
+            val = dictionary[key]
+            # Update the key value for this module if it already exists
+            db.module_values.update_or_insert((db.module_values.name == key) & (db.module_values.modulename == self.procedure_name),
+                                              time_stamp=datetime.datetime.utcnow(),
+                                              modulename=self.procedure_name,
```

Comment:
time_stamp might change in the DB table after merge with synchronization branch. It's good for now, but do remember to modify the field names once their branch is merged.